pkg/entrypoint/options.go

```diff
@@ -0,0 +1,115 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package entrypoint
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"time"
+
+	"github.com/knative/build/pkg/entrypoint/wrapper"
+)
+
+// NewOptions returns an empty Options with no nil fields
+func NewOptions() *Options {
+	return &Options{
+		Options: &wrapper.Options{},
+	}
+}
+
+// Options exposes the configuration necessary
+// for defining the process being watched and
+// where in the image repository an upload will land.
+type Options struct {
+	// Args is the process and args to run
+	Args []string `json:"args"`
+	// Timeout determines how long to wait before the
+	// entrypoint sends SIGINT to the process
+	Timeout time.Duration `json:"timeout"`
+	// GracePeriod determines how long to wait after
+	// sending SIGINT before the entrypoint sends
+	// SIGKILL.
+	GracePeriod time.Duration `json:"grace_period"`
+	// ArtifactDir is a directory where test processes can dump artifacts
+	// for upload to persistent storage (courtesy of sidecar).
+	// If specified, it is created by entrypoint before starting the test process.
+	// May be ignored if not using sidecar.
+	ArtifactDir string `json:"artifact_dir,omitempty"`
+
+	*wrapper.Options
+}
+
+// Validate ensures that the set of options are
+// self-consistent and valid
+func (o *Options) Validate() error {
+	if len(o.Args) == 0 {
+		return errors.New("no process to wrap specified")
+	}
+
+	return o.Options.Validate()
+}
+
+const (
+	// JSONConfigEnvVar is the environment variable that
+	// utilities expect to find a full JSON configuration
+	// in when run.
+	JSONConfigEnvVar = "ENTRYPOINT_OPTIONS"
+)
+
+// ConfigVar exposes the environment variable used
+// to store serialized configuration
+func (o *Options) ConfigVar() string {
+	return JSONConfigEnvVar
+}
+
+// LoadConfig loads options from serialized config
+func (o *Options) LoadConfig(config string) error {
+	return json.Unmarshal([]byte(config), o)
+}
+
+// AddFlags binds flags to options
+func (o *Options) AddFlags(flags *flag.FlagSet) {
+	flags.DurationVar(&o.Timeout, "timeout",
+		DefaultTimeout, "Timeout for the test command.")
+	flags.DurationVar(&o.GracePeriod, "grace-period",
+		DefaultGracePeriod, "Grace period after timeout for the test command.")
+	flags.StringVar(&o.ArtifactDir, "artifact-dir",
+		"", "directory where test artifacts should be placed for upload "+
+			"to persistent storage")
+	flags.BoolVar(&o.ShouldWaitForPrevStep, "should-wait-for-prev-step",
+		DefaultShouldWaitForPrevStep, "If we should wait for prev step.")
+	flags.BoolVar(&o.ShouldRunPostRun, "should-run-post-run",
+		DefaultShouldRunPostRun, "If post run step should be run.")
+	flags.StringVar(&o.PreRunFile, "prerun-file",
+		DefaultPreRunFile, "The prerun file to wait for.")
+	flags.StringVar(&o.PostRunFile, "postrun-file",
+		DefaultPostRunFile, "If postrun file to write.")
```

Comment:
can you explain more about what "file to wait for" means here? (e.g. the process will wait for this file to exist before xyz)